.golangci.yml

```diff
@@ -1,34 +1,43 @@
-run:
-  timeout: 10m
+version: "2"
 
 linters:
+  default: none
   enable:
     - depguard
-    - gofmt
-    - goimports
     - revive
     - govet
     - importas
     - ineffassign
     - misspell
-    - typecheck
     - errname
     - makezero
     - whitespace
-  disable-all: true
-
-linters-settings:
-  depguard:
+  settings:
+    depguard:
+      rules:
+        main:
+          deny:
+            - pkg: io/ioutil
+              desc: The io/ioutil package has been deprecated, see https://go.dev/doc/go1.16#ioutil
+    importas:
+      no-unaliased: true
+  exclusions:
+    generated: lax
+    presets:
+      - comments
+      - common-false-positives
+      - legacy
+      - std-error-handling
```

Comment:
ISTR specifying these makes it skip some (not run "more")? Wondering if CY is happy if presents is empty?

Reply:
Ah true and it fails for `comments` preset:

```
#20 [lint 1/1] RUN --mount=type=bind,target=.   --mount=type=cache,target=/root/.cache   --mount=from=golangci-lint,source=/usr/bin/golangci-lint,target=/usr/bin/golangci-lint   golangci-lint run ./...      
#20 0.995 annotation/annotation.go:15:1: package-comments: should have a package comment (revive)
#20 0.995 package annotation
#20 0.995 ^
#20 0.995 clidocstool.go:15:1: package-comments: should have a package comment (revive)
#20 0.995 package clidocstool
#20 0.995 ^
#20 0.995 2 issues:
#20 0.995 * revive: 2
#20 ERROR: process "/bin/sh -c golangci-lint run ./..." did not complete successfully: exit code: 1
```

I guess we can just exclude this one as it was not catched before?

Reply:
Ah, yes, either we can add a package comment (if there's only a few) or we can add excludes for a couple; https://github.com/moby/moby/blob/7937f0846c130e91836f5f640c41ce12d07da4c8/.golangci.yml#L152-L158

Reply:
> Ah, yes, either we can add a package comment (if there's only a few) or we can add excludes for a couple; https://github.com/moby/moby/blob/7937f0846c130e91836f5f640c41ce12d07da4c8/.golangci.yml#L152-L158

Is this one complementary or a duplicate https://github.com/moby/moby/blob/7937f0846c130e91836f5f640c41ce12d07da4c8/.golangci.yml#L146-L150 of staticcheck `- -ST1000`? Here seems only a revive rule.

Reply:
Added comments to packages